src/Analyzers/CSharp/CodeFixes/UseUTF8StringLiteral/UseUTF8StringLiteralCodeFixProvider.cs

```diff
@@ -0,0 +1,194 @@
+                }
+                else
+                {
+                    editor.ReplaceNode(node, CreateUTF8String(node, stringValue));
+                }
+            }
+        }
+
+        private static string GetUTF8StringValueForDiagnostic(SemanticModel semanticModel, Diagnostic diagnostic, CancellationToken cancellationToken)
+        {
+            // For computing the UTF8 string we need the original location of the array creation
+            // operation, which is stored in additional locations.
+            var location = diagnostic.AdditionalLocations[0];
+            var node = location.FindNode(getInnermostNodeForTie: true, cancellationToken);
+
+            var operation = semanticModel.GetRequiredOperation(node, cancellationToken);
+
+            var operationLocationString = diagnostic.Properties[nameof(UseUTF8StringLiteralDiagnosticAnalyzer.ArrayCreationOperationLocation)];
+            if (!Enum.TryParse(operationLocationString, out UseUTF8StringLiteralDiagnosticAnalyzer.ArrayCreationOperationLocation operationLocation))
+                throw ExceptionUtilities.Unreachable;
+
+            IArrayCreationOperation arrayOp;
+
+            // Because we get the location from an IOperation.Syntax, sometimes we have to look a
+            // little harder to get back from syntax to the operation that triggered the diagnostic
+            if (operationLocation == UseUTF8StringLiteralDiagnosticAnalyzer.ArrayCreationOperationLocation.Ancestors)
+            {
+                // For collection initializers where the Add method takes a param array, and the array creation
+                // will be a parent of the operation
+                arrayOp = FindArrayCreationOperationAncestor(operation);
+            }
+            else if (operationLocation == UseUTF8StringLiteralDiagnosticAnalyzer.ArrayCreationOperationLocation.Descendants)
+            {
+                // Otherwise, we must have an implicit array creation for a parameter array, so the location
+                // will be the invocation, or similar, that has the argument, and we need to descend child
+                // nodes to find the one we are interested in. To make sure we're finding the right one,
+                // we can use the diagnostic location for that, since the analyzer raises it on the first element.
+                arrayOp = operation.DescendantsAndSelf()
+                    .OfType<IArrayCreationOperation>()
+                    .Where(a => a.Initializer?.ElementValues.FirstOrDefault()?.Syntax.SpanStart == diagnostic.Location.SourceSpan.Start)
+                    .First();
+            }
+            else
+            {
+                arrayOp = (IArrayCreationOperation)operation;
+            }
+
+            Contract.ThrowIfNull(arrayOp.Initializer);
+
+            // Get our list of bytes from the array elements
+            using var _ = PooledStringBuilder.GetInstance(out var builder);
```

Comment:
nit, consider allocating with the number of elements in the array.  that will give a stringbuilder that will almost always not have to reallocate as you add elements.